lib/engine/game/g_1871/step/buy_sell_par_split_shares.rb

```diff
@@ -0,0 +1,174 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/buy_sell_par_shares'
+
+module Engine
+  module Game
+    module G1871
+      module Step
+        class BuySellParSplitShares < Engine::Step::BuySellParShares
+          def actions(entity)
+            return [] unless entity == current_entity
+            return ['sell_shares'] if must_sell?(entity)
+
+            actions = []
+            actions << 'buy_shares' if can_buy_any?(entity) || can_exchange_any?(entity)
+            actions << 'par' if can_ipo_any?(entity)
+            actions << 'split' if can_split_any?(entity)
+            actions << 'sell_shares' if can_sell_any?(entity)
+
+            actions << 'pass' unless actions.empty?
+            actions
+          end
+
+          def process_buy_shares(action)
+            super
+
+            @round.bank_bought = true if action.purchase_for == @game.union_bank
+          end
+
+          def can_buy_for(entity)
+            return [] unless entity == @game.company_by_id('UB').owner
+            return [] if @round.bank_bought
+
+            [@game.union_bank]
+          end
+
+          # This makes sure we don't auto pass if the player is allowed to
+          # exchange a share
+          def can_exchange_any?(entity)
+            return false unless entity.player?
+
+            entity.companies.each do |company|
+              return true if ["MC", "VR", "SB"].include?(company.id)
+
+              if company.id == "IB"
+                # Only true if there are shares available from an ipo'd company
+                # that isn't PEIR, Mainline or Shortline
+                @game.corporations.select(&:ipoed).reject do |c|
+                  @game.mainline == c or @game.shortline == c or @game.peir == c
+                end.each do |c|
```

Comment:
don't like this changed syntax with do end blocks

Reply:
Rubocop didn't either, Fixed